src/Identifier/Normalizer/ChainIdentifierNormalizer.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Identifier\Normalizer;
+
+use ApiPlatform\Core\Exception\InvalidIdentifierException;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Identifier normalizer.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class ChainIdentifierNormalizer implements IdentifierNormalizerInterface
+{
+    private $propertyNameCollectionFactory;
+    private $propertyMetadataFactory;
+    private $compositeIdentifierNormalizer;
+    private $identifierNormalizers;
+
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, array $identifierNormalizers)
+    {
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+        $this->compositeIdentifierNormalizer = new CompositeIdentifierNormalizer();
+        $this->identifierNormalizers = $identifierNormalizers;
+    }
+
+    public function normalize(string $identifier, string $resourceClass): array
+    {
+        $identifiers = [];
+        $metadataIdentifiers = [];
+        $identifiersNames = [];
+        $identifiersCount = 0;
+
+        // Get metadata identifiers
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $propertyName) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $propertyName);
+
+            if (!$propertyMetadata->isIdentifier()) {
+                continue;
+            }
+
+            $type = $propertyMetadata->getType();
+            if ($type && Type::BUILTIN_TYPE_OBJECT === $type->getBuiltinType()) {
+                $metadataIdentifiers[$propertyName] = $type->getClassName();
+            } else {
+                $metadataIdentifiers[$propertyName] = null;
+            }
+
+            $identifiersNames[] = $propertyName;
+            ++$identifiersCount;
+        }
+
+        // Convert to identifiers array
+        if (1 === $identifiersCount) {
+            $identifiers[$identifiersNames[0]] = $identifier;
+        } else {
+            $identifiers = $this->compositeIdentifierNormalizer->normalize($identifier);
```

Comment:
To me it should be a standard normalizer, that will call recursively other ones if needed. (as we do for data normalizers)